docs/src/solvers/index.md

```diff
@@ -1,43 +1,114 @@
 
-# List of available solvers
+# Available solvers in Manopt.jl
 
 ```@meta
 CurrentModule = Manopt
 ```
 
-Solvers can be applied to [`AbstractManoptProblem`](@ref)s with solver
-specific [`AbstractManoptSolverState`](@ref).
+Optimisation problems can be classified with respect to several criteria.
+In the following we provide a grouping of the algorithms with respect to the “information”
+available about your optimisation problem
 
-# List of algorithms
+```math
+\operatorname*{arg\,min}_{p∈\mathbb M} f(p)
+```
+
+Within the groups we provide short notes on advantages of the individual solvers,
+for example whether ``f`` should be convex, or nonconvex – or whether the method works
+locally or even globally.
+
+## Derivative Free
+
+For derivative free only function evaluations of ``f`` are used.
+
+* [Nelder-Mead](NelderMead.md) a simplex based variant, that is using ``d+1`` points, where ``d`` is the dimension of the manifold.
+* [Particle Swarm](particle_swarm.md) use the evolution of a set of points, called swarm, to explore the domain of the cost and find a minimizer.
+
+## First Order
+
+### Gradient
+
+* [Gradient Descent](gradient_descent.md) uses the gradient from ``f`` to determine a descent direction. Here, the direction can also be changed to be Averaged, Momentum-based, based on Nesterovs rule.
+* [Conjugate Gradient Descent](conjugate_gradient_descent.md) uses information from the previous descent direction to improve the current (gradient-based) one including several such update rules.
+* [Levenberg-Marquardt](LevenbergMarquardt.md) minimizes the square norm of ``f: \mathcal M→ℝ^d`` provided the gradients of the component functions, or in other words the Jacobian of ``f``.
+* The [Quasi-Newton Method](quasi_Newton.md) uses gradient evaluations to approximate the Hessian, which is then used in a Newton-like scheme, where both a limited memory and a full Hessian approximation are available with several different update rules.
+* [Steihaug-Toint Truncated Conjugate-Gradient Method](@ref tCG) a solver for a constrained problem defined on a tangent space.
+* [Stochastic Gradient Descent](stochastic_gradient_descent.md) is based on a splitting of ``f`` into several components ``f_i`` whose gradients are provided here. By randomly selecting these, steps towards (compnent-wise) descent directions are performed.
```

Comment:
```suggestion
* [Stochastic Gradient Descent](stochastic_gradient_descent.md) is based on a splitting of ``f`` into a sum of several components ``f_i`` whose gradients are provided. Steps are performed according to gradients of randomly selected components.
```
I think ideally the interface should be such that the user provides a single function but it gets component number `i` as an argument.

Also, isn't this technically a splitting method?

Reply:
Yes, technically both LM and SGD are splitting methods.

Reply:
Maybe one could rename the spitting section to “Nonsmooth Splitting methods”? That would be a remedy maybe that unifies what I collected there.